ts/screens/wallet/TransactionDetailsScreen.tsx

```diff
@@ -124,12 +125,17 @@ class TransactionDetailsScreen extends React.Component<Props> {
   private getData = () => {
     const transaction = this.props.navigation.getParam("transaction");
     const amount = formatNumberCentsToAmount(transaction.amount.amount, true);
-    const fee = formatNumberCentsToAmount(
-      transaction.fee === undefined
-        ? transaction.grandTotal.amount - transaction.amount.amount
-        : transaction.fee.amount,
-      true
-    );
+
+    // fee
+    const maybeFee = maybeInnerProperty<Transaction, "fee", number | undefined>(
+      transaction,
+      "fee",
+      m => (m !== undefined ? m.amount : 0)
```

Comment:
```suggestion
      m => (m ? m.amount : undefined)
```
this is should be a further improvement

Reply:
don't know why i didn't change it!

fixed in 440faae